test/plugin.test.ts

```diff
@@ -0,0 +1,79 @@
+import { beforeEach, describe, expect, it, TestOptions } from 'vitest';
+import scalaJSPlugin, { ScalaJSPluginOptions } from "../index";
+import type { PluginContext } from 'rollup';
+import type { Plugin as VitePlugin } from "vite";
+
+/* This interface refines the VitePlugin with some knowledge about our
+ * particular implementation, for easier testing. If we define here something
+ * that is incompatible with what's in VitePlugin, the compiler complains, so
+ * we do get some safety that we adhere to VitePlugin's API.
+ */
+interface RefinedPlugin extends VitePlugin {
+  buildStart: (this: PluginContext, options: {}) => void | Promise<void>;
+  resolveId: (this: PluginContext, source: string) => string | Promise<string>;
+}
+
+function normalizeSlashes(path: string | null): string | null {
+  return path === null ? null : path.replace(/\\/g, '/');
+}
+
+describe("scalaJSPlugin", () => {
+  const cwd = process.cwd() + "/test/testproject";
+
+  const testOptions: TestOptions = {
+    timeout: 60000, // running sbt takes time
+  };
+
+  const setup: (options: ScalaJSPluginOptions) => [RefinedPlugin, PluginContext] = (options) => {
+    const plugin = scalaJSPlugin({ cwd: cwd, ...options }) as RefinedPlugin;
+    const fakePluginContext = {} as PluginContext;
+    return [plugin, fakePluginContext];
+  }
+
+  /* Wait for 2 seconds between tests to let sbt close its server.
+   * Without this, we get spurious failures with
+   * > sbt thinks that server is already booting because of this exception:
+   * > sbt.internal.ServerAlreadyBootingException: java.io.IOException:
+   *   Could not create lock for \\.\pipe\sbt-load3101661995253037154_lock, error 5
+   */
+  beforeEach(() => {
+    return new Promise(resolve => {
+      setTimeout(resolve, 2000);
+    });
+  });
+
+  it("works without a specific projectID", async () => {
+    const [plugin, fakePluginContext] = setup({});
+
+    await plugin.buildStart.call(fakePluginContext, {});
+
+    expect(normalizeSlashes(await plugin.resolveId.call(fakePluginContext, 'scalajs:main.js')))
+      .toContain('/testproject/target/scala-3.2.2/testproject-opt/main.js');
```

Comment:
Have you considered checking for strict equality using `path.join`:

```ts
const expected = path.join(cwd, "target", "scala-3.2.2", "testproject-opt", "main.js")
```

Reply:
The Vite documentation says that it internally normalizes paths to using slashes, but it's unclear when and how. It seems more consistent to me to write tests with `/` everywhere, to be closer to that idea.